libvast/vast/bloom_filter.hpp

```diff
@@ -65,11 +70,25 @@ class bloom_filter : detail::equality_comparable<
 
   /// Adds an element to the Bloom filter.
   /// @param x The element to add.
+  /// @returns `void` if the tracking policy is `tracking::no`,
+  /// otherwise a `bool` that is `false` iff *x* already exists in the filter.
   template <class T>
-  void add(T&& x) {
+  auto add(T&& x) {
```

Comment:
Please `[[nodiscard]]` this one.

Reply:
But this can be `void` as well. Is this well-defined?

It's also entirely optionally signaling.

Reply:
The optional signaling is opt-in, so if users enable that I assume they also want to catch this.

The nodiscard is well-defined on void.